src/data/markdown/translated-guides/en/02 Using k6/05 k6 Options/01 How to.md

```diff
@@ -0,0 +1,166 @@
+excerpt: 'Examples of different ways to set options. Options in command-line flags have the highest precedence. You can also access option values as the test runs.'
+---
+
+Your use case will likely determine how you want to set options.
+You can also access option values as your test runs.
+
+## Where to set options
+
+For most options, k6 provides multiple ways to set them:
+
+- In CLI flags
+- In environment variables
+- In the config file
+- In the script `options` object
+
+You can also set an option in one place and override it somewhere else (according to the [order of precedence](../#order-of-precedence)).
+Sometimes, how you set options is a matter of personal preference.
+Other times, your situation will dictate the best place to put your options.
+
+*CLI options are good setting options on the fly.*
+
+Of course, you can configure options in multiple places.
+For example, you might want to set your stages in the script options, then use CLI flags to run the script with different duration times.
+
+*Environment variables often work well when you need to set your options from some other part of your DevOps build chain*.
+
+For example, you could derive the option from a variable in your Docker container definition, CI UI, vault&mdash;wherever you declare environment variables.
+The [block hostnames](#block-hostnames) option is an option that works well with environment variables.
+
+*Options in the script or config file let you version control your options (and keep your tests tidy).*
+
+You may wonder why you would need a config file when you can use the script options.
+Isn't this redundant?
+In fact, you might have multiple reasons to set your options in a config file.
+
+For one, k6 uses the (auto-generated) config file to keep users logged in after they run `k6 login cloud`.
+Other reasons for config files are more about developer style and comfort:
+- To separate the test configuration from JavaScript code
+- To reduce the size of the entry-point script, especially when there are multiple scenarios
+- To more easily switch between configs (either rename the current config file or modify the JavaScript code in one location)
+
+
+## Examples of setting options
+
+The following JS snippets show some examples of how you can set options.
+
+### Options in the script
+
+<CodeGroup labels={["example.js"]} lineNumbers={[true]}>
+
+```javascript
+import http from 'k6/http';
+
+export const options = {
+  hosts: { 'test.k6.io': '1.2.3.4' },
+  stages: [
+    { duration: '1m', target: 10 },
+    { duration: '1m', target: 20 },
+    { duration: '1m', target: 0 },
+  ],
+  thresholds: { http_req_duration: ['avg<100', 'p(95)<200'] },
+  noConnectionReuse: true,
+  userAgent: 'MyK6UserAgentString/1.0',
+};
+
+export default function () {
+  http.get('http://test.k6.io/');
+}
+```
+
+</CodeGroup>
+
+### Options in a config file
+
+<div id="config-json-example">
+You can also define the same options through a config file:
```

Comment:
Creating `config.json` file is not enough. We must show how this config is read in the script.

```
export const options = open('config.json')
```